NETOBSERV-1793 Introduce ASCIIdoc for CLI (#85)
netobserv/network-observability-cli: docs/netobserv_cli.adoc

// Automatically generated by './scripts/generate-doc.sh'. Do not edit, or make the NETOBSERV team aware of the editions.
:_mod-docs-content-type: REFERENCE
[id="network-observability-cli-usage_{context}"]
= Network Observability CLI usage

You can use the The Network Observability CLI (`oc netobserv`) to pass command line arguments to capture flow data and packet data for further analysis, enable Network Observability Operator features or pass configuration options to the eBPF agent and flowlogs-pipeline.
== Syntax 

> The basic syntax for `oc netobserv` commands: 

Reviewer: I think you need a carriage return between 8 and 9, otherwise the rendering is wonky

Author: https://github.com/netobserv/network-observability-cli/pull/85/commits/38470dbfd5b30f5e94a68873d5f32f1e4c76e999

|--direction|       flow filter direction               | n/a
|--cidr|            flow filter CIDR                    | 0.0.0.0/0
|--protocol|        flow filter protocol                | n/a
|--sport|           flow filter source port             | n/a
|--dport|           flow filter destination port        | n/a
|--port|            flow filter port                    | n/a
|--sport_range|     flow filter source port range       | n/a
|--dport_range|     flow filter destination port range  | n/a
|--port_range|      flow filter port range              | n/a
|--tcp_flags|       flow filter TCP flags               | n/a
|--icmp_type|       ICMP type                           | n/a
|--icmp_code|       ICMP code                           | n/a
|--peer_ip|         peer IP                             | n/a
|--action|          flow filter action                  | Accept
|--interfaces|      interfaces to monitor               | n/a
|===

.Example running flow capture with options:
[source,terminal]
----
$ oc netobserv flows --enable_pktdrop=true  --enable_rtt=true --enable_filter=true --action=Accept --cidr=0.0.0.0/0 --protocol=TCP --port=49051
----
== Packets capture options

> You can filter on port and protocol for packet capture data. 
> .`oc netobserv packets` syntax

Reviewer: ```suggestion
You can filter on port and protocol for packet capture data. 

.`oc netobserv packets` syntax
```

Reviewer: Add a carriage return here to fix the rendering issues

Author: https://github.com/netobserv/network-observability-cli/pull/85/commits/38470dbfd5b30f5e94a68873d5f32f1e4c76e999

// Automatically generated by './scripts/generate-doc.sh'. Do not edit, or make the NETOBSERV team aware of the editions.
:_mod-docs-content-type: REFERENCE
[id="network-observability-cli-usage_{context}"]
= Network Observability CLI usage

> Users of Netobserv CLI can pass command line arguments to enable features or pass configuration options to the eBPF agent and flowlogs-pipeline.

Reviewer: ```suggestion
You can use the The Network Observability CLI (`oc netobserv`) to pass command line arguments to capture flow data and packet data for further analysis, enable Network Observability Operator features or pass configuration options to the eBPF agent and flowlogs-pipeline.
```

Reviewer: Can we add more of these details we have here: https://docs.openshift.com/container-platform/4.12/observability/network_observability/netobserv_cli/netobserv-cli-reference.html

This document lists all the supported command line options and their possible values.

> 

Reviewer: == Basic commands

[cols="3a,8a",options="header"]
.Basic commands
|===
|Command| Description

| `flows`
| Capture flows information. For subcommands, see the "Flow capture subcommands" table. 

| `packets`
| Capture packets from a specific protocol or port pair, such as `netobserv packets --filter=tcp,80`. For more information about packet capture, see the "Packet capture subcommand" table.

| `cleanup`
| Remove the Network Observability CLI components.

| `version`
| Print the software version.

| `help`
| Show help.
|===

Reviewer: Even though you can easily print this in the terminal, we should still add it for the sake of being thorough.

> 

Reviewer: ```suggestion

== Basic commands

[cols="3a,8a",options="header"]
.Basic commands
|===
|Command| Description

| flows
| Capture flows information. For subcommands, see the "Flow capture subcommands" table.

| packets
| Capture packets from a specific protocol or port pair, such as netobserv packets --filter=tcp,80. For more information about packet capture, see the "Packet capture subcommand" table.

| cleanup
| Remove the Network Observability CLI components.

| version
| Print the software version.

| help
| Show help.
|===

```

> == Flows usage

Reviewer: ```suggestion
== Flows capture options
```

Reviewer: This is what we have it titled as currently

[cols="1,1,1",options="header"]
|===
| Option | Description | Default
|--enable_pktdrop|  enable packet drop                  | false
|--enable_dns|      enable DNS tracking                 | false

> |--enable_rtt|      enable RTT tracking                 | false

Reviewer: Currently, we have these called out as enrichment options along with `interfaces` and `help`. Maybe that distinction isn't necessary, but I wanted to point it out and discuss. https://docs.openshift.com/container-platform/4.12/observability/network_observability/netobserv_cli/netobserv-cli-reference.html#network-observability-cli-enrichment_netobserv-cli-reference

Author: I'm not sure it's mandatory to make the distinction between filters and features (pktdrop / dns / rtt) but I can improve the script in that way if you feel it was better.

Interfaces is a filter only available for flows so it has been moved to the proper place.
`--help` option has been replaced by `help` sub command as `oc netobserv flows help`

Reviewer: Okay, if it isn't mandatory then we don't have to make that distinction.

|--max-time|        maximum capture time                | 5m
|--max-bytes|       maximum capture bytes               | 50000000 = 50MB
|--copy|            copy the output files locally       | prompt
|--direction|       flow filter direction               | n/a
|--cidr|            flow filter CIDR                    | 0.0.0.0/0
|--protocol|        flow filter protocol                | n/a
|--sport|           flow filter source port             | n/a
|--dport|           flow filter destination port        | n/a
|--port|            flow filter port                    | n/a
|--sport_range|     flow filter source port range       | n/a
|--dport_range|     flow filter destination port range  | n/a
|--port_range|      flow filter port range              | n/a
|--tcp_flags|       flow filter TCP flags               | n/a
|--icmp_type|       ICMP type                           | n/a
|--icmp_code|       ICMP code                           | n/a
|--peer_ip|         peer IP                             | n/a
|--action|          flow filter action                  | Accept
|--interfaces|      interfaces to monitor               | n/a
|===

> Example running flow capture with some options:

Reviewer: I like the idea of putting an example too. The `.` just adds a formatting, so it looks like the heading of other examples:
![image](https://github.com/user-attachments/assets/2761e606-20ca-474f-820d-6795f99424b9)

> ```
> oc netobserv flows --enable_pktdrop=true  --enable_rtt=true --enable_filter=true --action=Accept --cidr=0.0.0.0/0 --protocol=TCP --port=49051
> ```

Reviewer: ```suggestion
[source,terminal]
----
$ oc netobserv flows --enable_pktdrop=true  --enable_rtt=true --enable_filter=true --action=Accept --cidr=0.0.0.0/0 --protocol=TCP --port=49051
----
```

Reviewer: In docs, we format terminal examples like this.

> == Packets usage

Reviewer: ```suggestion
== Packets capture options
You can filter on port and protocol for packet capture data. 

.`oc netobserv packets` syntax
[source,terminal]
----
$ oc netobserv packets [<option>]
----
```

Reviewer: I would like to retain some intro text for the table. I guess packet capture is more beefy in this release so its more than just port and protocol filtering.

|===
| Option | Description | Default
|--log-level|       components logs                     | info
|--max-time|        maximum capture time                | 5m
|--max-bytes|       maximum capture bytes               | 50000000 = 50MB
|--copy|            copy the output files locally       | prompt
|--direction|       flow filter direction               | n/a
|--cidr|            flow filter CIDR                    | 0.0.0.0/0
|--protocol|        flow filter protocol                | n/a
|--sport|           flow filter source port             | n/a
|--dport|           flow filter destination port        | n/a
|--port|            flow filter port                    | n/a
|--sport_range|     flow filter source port range       | n/a
|--dport_range|     flow filter destination port range  | n/a
|--port_range|      flow filter port range              | n/a
|--tcp_flags|       flow filter TCP flags               | n/a
|--icmp_type|       ICMP type                           | n/a
|--icmp_code|       ICMP code                           | n/a
|--peer_ip|         peer IP                             | n/a
|--action|          flow filter action                  | Accept
|===

> Example running packet capture with some options:

Reviewer: ```suggestion
.Example running packet capture with options:
```

> ```
> oc netobserv packets --action=Accept --cidr=0.0.0.0/0 --protocol=TCP --port=49051
> ```

Reviewer: ```suggestion
[source,terminal]
----
$ oc netobserv packets --action=Accept --cidr=0.0.0.0/0 --protocol=TCP --port=49051
----
```
